app/code/Magento/QuoteGraphQl/Model/Resolver/MergeCarts.php

```diff
@@ -31,44 +37,93 @@ class MergeCarts implements ResolverInterface
      */
     private $cartRepository;
 
+    /**
+     * @var CustomerCartResolver
+     */
+    private $customerCartResolver;
+
+    /**
+     * @var QuoteIdToMaskedQuoteIdInterface
+     */
+    private $quoteIdToMaskedQuoteId;
+
     /**
      * @param GetCartForUser $getCartForUser
      * @param CartRepositoryInterface $cartRepository
+     * @param CustomerCartResolver $customerCartResolver
+     * @param QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
      */
     public function __construct(
         GetCartForUser $getCartForUser,
-        CartRepositoryInterface $cartRepository
+        CartRepositoryInterface $cartRepository,
+        CustomerCartResolver $customerCartResolver,
+        QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
     ) {
         $this->getCartForUser = $getCartForUser;
         $this->cartRepository = $cartRepository;
+        $this->customerCartResolver = $customerCartResolver;
+        $this->quoteIdToMaskedQuoteId = $quoteIdToMaskedQuoteId;
     }
 
     /**
      * @inheritdoc
      */
-    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
-    {
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
         if (empty($args['source_cart_id'])) {
-            throw new GraphQlInputException(__('Required parameter "source_cart_id" is missing'));
-        }
-
-        if (empty($args['destination_cart_id'])) {
```

Comment:
Just a suggestion. If the `destination_cart_id` is not set, most likely we will have the "Undefined index" notice. We may check if it is set, and then check it is empty.